src/OrchardCore/Orchard.Data.Abstractions/ServiceCollectionExtensions.cs

```diff
@@ -1,27 +1,24 @@
 ﻿using System;
+using System.Linq;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace Orchard.Data
 {
     public static class ServiceCollectionExtensions
     {
-        public static IServiceCollection TryAddDataProvider(this IServiceCollection services, string name, string value, bool hasConnectionString)
+        public static IServiceCollection TryAddDataProvider(this IServiceCollection services, string name, string value, bool hasConnectionString = false)
         {
-            for (var i = services.Count - 1; i >= 0; i--)
+            var serviceProvider = services.BuildServiceProvider();
```

Comment:
No way ;) No refactoring can justify the need to build the container early like this.